minio-client/main.go

```diff
@@ -0,0 +1,115 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strconv"
+	"time"
+
+	minio "github.com/minio/minio-go"
+)
+
+var (
+	accessKeyID              = os.Getenv("ACCESS_KEY")
+	secretAccessKey          = os.Getenv("SECRET_KEY")
+	serviceName              = os.Getenv("SERVICE_NAME")
+	namespace                = os.Getenv("NAMESPACE")
+	port                     = os.Getenv("PORT")
+	livenessCheckInterval, _ = strconv.Atoi(os.Getenv("LIVENESS_CHECK_INTERVAL"))
+	livenessRetryCount, _    = strconv.Atoi(os.Getenv("LIVENESS_RETRY_COUNT"))
+	mode                     = os.Getenv("MODE")
+	capacity, _              = strconv.ParseFloat(os.Getenv("CAPACITY"), 32)
+)
+
+func main() {
+	fileName := "bigfile.test"
+	bucketName := "test-bucket" + "-" + fmt.Sprintf("%d", time.Now().Unix())
+	endpoint := serviceName + "." + namespace + ".svc.cluster.local:" + port
+	useSSL := false
+
+	// Initialize minio client object.
+	minioClient, err := minio.New(endpoint, accessKeyID, secretAccessKey, useSSL)
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	// log.Printf("%#v\n", minioClient) // minioClient is now setup
+
+	createBucket(minioClient, bucketName, "us-east-1")
+	if mode == "liveness" {
+		createContent(fileName, 2097152)
+		livenessCheck(minioClient, bucketName, fileName)
+	} else {
+		fileSize := int(capacity * 1024 * 1024 * 1024)
+		createContent(fileName, fileSize)
+		fmt.Println("Load generation may take some time")
+		loadBucket(minioClient, bucketName, fileName)
+	}
+
+}
+
+func createBucket(client *minio.Client, bucketName string, region string) {
+	err := client.MakeBucket(bucketName, region)
+	if err != nil {
+		fmt.Println(err)
```

Comment:
why not fatal here? also this could be indicative of an unavailable minio endpoint

Reply:
I have made changes, now the very first time when bucket will be created and if it fails it will retry.

Reply:
ok. So, you plan to keep initRetry and LivenessRetry as same. Which is also OK.